### DISCUSSION TITLE
NavigationLink with Tags and Selection not working in ios 16

### DISCUSSION THREAD

**Discussion by knight6700** [pointfreeco/swift-composable-architecture#1578] [Q&A]
My Navigation Link it was working on ios 15 but now not working on ios 16 
i need to know which practice to implement Routing in swiftui

```
            NavigationLink(destination:
                Text("Profile"),
                tag: MenuRoute.Tag.profile,
                selection: viewStore.binding(get: \.tag, send: ProfileHeader.Action.navigateToProfile(tag:))) {
                    Image.horizon(MenuImage.profileIcon)
                        .resizable()
                        .frame(width: .horizonSpacing(.xl),
                               height: .horizonSpacing(.xl))
                        .background(.white)
                        .roundedEdge(width: 1,
                                     color: .horizon(.borderLineColor),
                                     cornerRadius: .horizonSpacing(.s))
                        .shadow(color: .black.opacity(0.06),
                                radius: 10, x: 0, y: 4)
                }

```

**Comment by stephencelis** [comment]
@knight6700 We're working on dedicated navigation tools at the moment, but for now this style should still work. Are you using `NavigationView` or `NavigationStack`? I'm not sure this deprecated style of `NavigationLink` is compatible with `NavigationStack`, so you may need to use the deprecated `NavigationLink` instead, or update to one of the new APIs if your application is iOS 16+.

**Reply by knight6700** [reply]
it's working but when i change tag to implement  a deep link not working on ios 16

**Reply by stephencelis** [reply]
@knight6700 Are you using `navigationDestination`? Do you think you're seeing this bug? https://gist.github.com/mbrandonw/f8b94957031160336cac6898a919cbb7#file-fb11056434-md

**Reply by knight6700** [reply]
you mean i should use `NavigationLink(value: Route.tag.Profile, label: ) ` with `. navigationDestination`

**Reply by stephencelis** [reply]
@knight6700 Sorry I'm just trying to understand the problem you face. Can you maybe share a full working project and describe what isn't working?

**Comment by abel3cl** [comment]
Hey, 

I have something similar. Using tag and destination in a NavigationLink.
I have replicated it in a smallish iOS app, you can find here https://github.com/abel3cl/NavigationSpike/tree/main/NavigationSpike

It's basically a List which contains some Rows(NavigationLinks) which themselves contains other NavigationLink
2 levels of navigation seems to be the problem, we're thinking probably in store.scope method.

**Reply by knight6700** [reply]
But my problem with 3 or 4 levels get back to root if make this in more than two levels

**Reply by abel3cl** [reply]
Do you have a possible solution/suggestion for this @stephencelis? Thanks
